lib/scraping/problem_checker.rb

```diff
@@ -0,0 +1,51 @@
+class ProblemChecker
+  @human_verified_time = 365
+  @outdated_time = 183
+
+  def data_check_routine
+    check_empty_fields(Person)
+    check_empty_fields(Room)
+    check_for_outdated(Room)
+  end
+
+  def check_empty_fields(model)
+    model.all.each do |entry|
+      entry.attributes.each do |name, value|
+        save_problem("missing", entry, name) unless value || name["human_verified"]
+      end
+    end
+  end
+
+  def check_for_outdated(model)
+    model.all.each do |entry|
+      save_problem("outdated", entry, "all") if entry.updated_at.days_since(@outdated_time).past?
+    end
+  end
+
+  def check_for_conflict(entry, new, field)
+    old = entry.public_send(field)
+    return true if !new || new == old
+
+    if !entry.public_send("human_verified_#{field}")
+      if entry.updated_at.days_since(1).future?
+        save_problem('conflicting', entry, field)
+        return true
+      end
+    elsif entry.public_send("human_verified_#{field}").days_since(@human_verified_time).past?
+      save_problem('outdated', entry, field)
+      return true
+    else
+      return true
```

Comment:
Why is there a conflict in this case?

Reply:
In this case human_verified is set for this field so a human approved this information and one year is not past yet since a human verified this information so we don`t want to overwrite and update that information, while also not saving a data_problem according to specifications.